tests/console/snapper_zypper.pm

```diff
@@ -0,0 +1,34 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Check that snapshot creation and rollback are not break rpmdb
+# Maintainer: Anton Smorodskyi <asmorodskyi@suse.com>
+
+use base 'consoletest';
+use testapi;
+use strict;
+use utils;
+use warnings;
+
+
+sub run {
+    my ($self) = @_;
+
+    select_console('root-console');
+    my $clean = script_output("snapper create -p -d clean");
+    my $pkgname = get_package_for_install();
+    zypper_call("in $pkgname");
+    assert_script_run("rpm -q $pkgname");
+
+    $self->rollback_and_reboot($clean);
+    assert_script_run("! rpm -q $pkgname");
+
+}
```

Comment:
If we want to check for inconsistency as Michal has found. We should check for file existence as well. 
One thing is state of rpmdb located `var` subvolume, and the other is existence of file provided by some package which is stored in other subvolume than `var`.

Reply:
Just 2 points, nothing to do with the functionality itself.

1. I would rather issue the command "as ordinary user" and evaluate on perl level the return code. 
2. Instead of creating new module, let's put the functionality to the original one. It would be nice to have all subvolumes tested in one module.

Reply:
> Just 2 points, nothing to do with the functionality itself.
> 
>     1. I would rather issue the command "as ordinary user" and evaluate on perl level the return code.

can you elaborate benefits which we getting from switching consoles several times ( snapper rollback , package installation still should happen under root ) ? 

> 
>     2. Instead of creating new module, let's put the functionality to the original one. It would be nice to have all subvolumes tested in one module.

not sure what you mean under "it would be nice" can you state some practical benefits ? From my perspective advantage of current approach is that you getting better granularity of results ( in case basic rollback is working but there are issues in btrfs setup you will clearly see this ) . disadvantage is more testing time but I don't see this as big issue. 

Also I have issue with "original module" which is `snapper_ jeos_cli` I would assume ? I can not figure out why it has name which contains "jeos_cli" . You suggested me to read Michal's commit messages but unfortunately this file initial commit does not state some useful info . From my POV `snapper_jeos_cli`  does not have any functionality which is specific to jeos so I would rename it to "snapper_snapshot" than it will make more sense to merge current code ( which also BTW is not jeos specific , right ? ) but still in this case we will loose granularity benefit which I mentioned above ...

Reply:
1. Is about the code, when you are testing existence of the package, file etc.
2. Subvolumes in JeOS are defined in kiwi template. So let's consider it as JeOS specific

Reply:
>     1. Is about the code, when you are testing existence of the package, file etc.

and why it is better to test existence of package / file as regular user ? 

> 
>     2. Subvolumes in JeOS are defined in kiwi template. So let's consider it as JeOS specific

ok let's leave point with renaming `snapper_jeos_cli` behind not that it is really critical. Anyway your initial point was about merging snapper_zypper into snapper_jeos_cli and I am not really like this idea , but I am trying to move our discussion to some practical dimension because 'like/dislike'  is dead-end

Reply:
I see the misunderstanding, as user interacting with cli/shell, do you write commands `! ...` ? OpenQA is basically simulating this interaction, that is also the reason why you have 4 time the same condition with 5 lines of code.

Reply:
> 
> ok let's leave point with renaming `snapper_jeos_cli` behind not that it is really critical. Anyway your initial point was about merging snapper_zypper into snapper_jeos_cli and I am not really like this idea , but I am trying to move our discussion to some practical dimension because 'like/dislike' is dead-end

Again, the argument with the subvolumes. Better to have 1 module which has the same purpose, rather than to split it into multiple modules without any real additional value.

Reply:
And again initial request was to extend `snapper_jeos_cli`, not to split it.

Reply:
> And again initial request was to extend `snapper_jeos_cli`, not to split it.

this is quote from https://progress.opensuse.org/issues/42353 : 

`The test module likely needs to be extended, or new test created. E.g via testing for (non-)existence of a particular RPM package:`

Reply:
not in conflict with the subvolume argument

Reply:
If you want to be picky in a child way subject `[qac][jeos] Extend snapper_jeos_cli test module` :D

And you missed this part as well. :D

```
Although we have snapper_jeos_cli test module which tests btrfs subvolumes on JeOS, we still found 'Bug 1111529 - kvm-and-xen image has incorrect subvolume configuration' (https://bugzilla.suse.com/show_bug.cgi?id=1111529) via manual testing.
```